positron/modules/atom_common_id_weak_map.js

```diff
@@ -0,0 +1,24 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use strict";
+
+const { classes: Cc, interfaces: Ci, results: Cr, utils: Cu } = Components;
+
+// Described In Electron's id_weak_map.h as:
+// "Like ES6's WeakMap, but the key is Integer and the value is Weak Pointer."
+
+function IDWeakMap() {
+  const map = new Map();
+  this.set = function(key, value) {
+    map.set(key, Cu.getWeakReference(value));
+  };
+  this.get = function(key) {
+    return map.get(key).get();
+  }
+  this.has = map.has.bind(map);
+  this.clear = map.clear.bind(map);
+}
```

Comment:
Do you know if it matters that these properties are on the instance rather than the prototype? If we want them on the prototype, we might need something like this:

```JS 
const instances = new WeakMap();

function IDWeakMap() {
  instances.set(this, new Map());
}

IDWeakMap.prototype = {
  set(key, value) {
    instances.get(this).set(key, Cu.getWeakReference(value));
    return this;
  },
  get(key) {
    return instances.get(this).get(key).get();
  },
  has(value) {
    return instances.get(this).has(value);
  },
  clear() {
    instances.get(this).clear();
  },
};

```